src/Template.cpp

```diff
@@ -81,15 +81,15 @@ CString CTemplateLoopContext::GetValue(const CString& sName, bool bFromIf) {
 }
 
 CTemplate::~CTemplate() {
-	for (map<CString, vector<CTemplate*> >::iterator it = m_mvLoops.begin(); it != m_mvLoops.end(); ++it) {
-		vector<CTemplate*>& vLoop = it->second;
-		for (unsigned int a = 0; a < vLoop.size(); a++) {
-			delete vLoop[a];
+	for (const auto& it : m_mvLoops) {
+		const vector<CTemplate*>& vLoop = it.second;
+		for (CTemplate* pTemplate : vLoop) {
+			delete pTemplate;
```

Comment:
It gets clumsy to keep backwards compatible ```std::vector<T*> GetXxx()``` -like methods if we store ```std::vector<std::shared_ptr<T>>``` internally. Are you willing to allow such API breakages? I'm also interested in #506 but that implies some bigger API changes as well.